jobs/resource_job_test.go

```diff
@@ -1419,6 +1419,113 @@ func TestResourceJobUpdate(t *testing.T) {
 	assert.Equal(t, "Featurizer New", d.Get("name"))
 }
 
+func TestResourceJobUpdate_NodeTypeToInstancePool(t *testing.T) {
+	d, err := qa.ResourceFixture{
+		Fixtures: []qa.HTTPFixture{
+			{
+				Method:   "POST",
+				Resource: "/api/2.1/jobs/reset",
+				ExpectedRequest: UpdateJobRequest{
+					JobID: 789,
+					NewSettings: &JobSettings{
+						NewCluster: &clusters.Cluster{
+							InstancePoolID:       "instance-pool-worker",
+							DriverInstancePoolID: "instance-pool-driver",
+							SparkVersion:         "spark-1",
+							NumWorkers:           1,
+						},
+						Tasks: []JobTaskSettings{
+							{
+								NewCluster: &clusters.Cluster{
+									InstancePoolID:       "instance-pool-worker-task",
+									DriverInstancePoolID: "instance-pool-driver-task",
+									SparkVersion:         "spark-2",
+									NumWorkers:           2,
+								},
+							},
+						},
+						JobClusters: []JobCluster{
+							{
+								NewCluster: &clusters.Cluster{
+									InstancePoolID:       "instance-pool-worker-job",
+									DriverInstancePoolID: "instance-pool-driver-job",
+									SparkVersion:         "spark-3",
+									NumWorkers:           3,
+								},
+							},
+						},
+						Name:                   "Featurizer New",
+						MaxRetries:             3,
+						MinRetryIntervalMillis: 5000,
+						RetryOnTimeout:         true,
+						MaxConcurrentRuns:      1,
+					},
+				},
+			},
+			{
+				Method:   "GET",
+				Resource: "/api/2.1/jobs/get?job_id=789",
+				Response: Job{
+					JobID: 789,
+					Settings: &JobSettings{
+						NewCluster: &clusters.Cluster{
+							NodeTypeID:       "node-type-id",
+							DriverNodeTypeID: "driver-node-type-id",
+						},
+						Name:                   "Featurizer New",
+						MaxRetries:             3,
+						MinRetryIntervalMillis: 5000,
+						RetryOnTimeout:         true,
+						MaxConcurrentRuns:      1,
+					},
+				},
+			},
+		},
+		ID:       "789",
+		Update:   true,
+		Resource: ResourceJob(),
+		InstanceState: map[string]string{
+			"new_cluster.0.node_type_id":                      "node-type-id",
+			"new_cluster.0.driver_node_type_id":               "driver-node-type-id",
+			"task.0.new_cluster.0.node_type_id":               "node-type-id-task",
+			"task.0.new_cluster.0.driver_node_type_id":        "driver-node-type-id-task",
+			"job_cluster.0.new_cluster.0.node_type_id":        "node-type-id-job",
+			"job_cluster.0.new_cluster.0.driver_node_type_id": "driver-node-type-id-job",
```

Comment:
micronit: let's align the names you use so they are consistent between node types and instance pools, just makes it a bit easier to see what corresponds to what
```suggestion
			"new_cluster.0.node_type_id":                      "node-type-id-worker",
			"new_cluster.0.driver_node_type_id":               "node-type-id-driver",
			"task.0.new_cluster.0.node_type_id":               "node-type-id-worker-task",
			"task.0.new_cluster.0.driver_node_type_id":        "node-type-id-driver-task",
			"job_cluster.0.new_cluster.0.node_type_id":        "node-type-id-worker-job",
			"job_cluster.0.new_cluster.0.driver_node_type_id": "node-type-id-driver-job",
```

Reply:
Hmm, I followed the pattern of how the attributes are named...

Reply:
It's fine, it is just slightly inconsistent with the instance pools literals you used. Or, vice-versa, you could rename the parameters "instance-pool-job" and "driver-instance-pool-job" in the request & HCL.